sdk/tcp_listener.go

```diff
@@ -14,10 +14,16 @@ const (
 )
 
 func newTCPListener(address string, pluginName string) (net.Listener, string, error) {
-	listener, err := sockets.NewTCPSocket(address, nil)
+	listener, err := setupSocketActivation()
```

Comment:
isn't this going to use a unix socket when you want a TCP one? it sounds like a bad behavior.